packages/synapse-react-client/src/components/EntityUpload/EntityUploadPromptDialog.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  FormControlLabel,
+  Typography,
+} from '@mui/material'
+import { noop } from 'lodash-es'
+import pluralize from 'pluralize'
+import React, { useState } from 'react'
+import {
+  PromptInfo,
+  UseUploadFileEntitiesReturn,
+} from '../../utils/hooks/useUploadFileEntity/useUploadFileEntities'
+import { DialogBase } from '../DialogBase'
+
+type EntityUploadPromptDialogProps = {
+  activePrompts: UseUploadFileEntitiesReturn['activePrompts']
+}
+
+function getTitle(promptInfo: PromptInfo) {
+  if (promptInfo.type === 'CONFIRM_NEW_VERSION') {
+    return 'Update existing file?'
+  }
+  return ''
+}
+
+function getMessage(promptInfo: PromptInfo) {
+  if (promptInfo.type === 'CONFIRM_NEW_VERSION') {
+    return `A file named "${promptInfo.fileName}" (${promptInfo.existingEntityId}) already exists in this location. Do you want to update the existing file and create a new version?`
+  }
+  return ''
+}
+
+export function EntityUploadPromptDialog(props: EntityUploadPromptDialogProps) {
+  const { activePrompts } = props
+  const [yesToAllChecked, setYesToAllChecked] = useState(false)
+
+  if (activePrompts.length === 0) {
+    return null
+  }
+
+  const numberOfOtherPrompts = activePrompts.length - 1
```

Comment:
Should this numberOfOtherPrompts filter by info.type (to CONFIRM_NEW_VERSION only)?  (for a future where there are other types)?